### PR TITLE
mgr/dashboard_v2: README update

### PR DESCRIPTION
**Pull request by rjfd**
Changes include:
* updated the whole unit tests content to match the latest changes on unit test execution
* updated the development notes about controller development

Signed-off-by: Ricardo Dias <rdias@suse.com>

### CODE REVIEW

File: src/pybind/mgr/dashboard_v2/README.rst [hunk 1]
```diff
@@ -71,30 +71,80 @@ Unit Testing and Linting
...
+
+Our ``tox.ini`` script will start a ``vstart`` Ceph cluster before running the
+python unit tests, and then it stops the cluster after the tests are run.
+Of course this implies that you have built/compiled Ceph previously.
 
 To run tox, run the following command in the root directory (where ``tox.ini``
 is located)::
 
-  $ tox
+  $ PATH=../../../../build/bin:$PATH tox
```
Comment (sebastian-philipp) [inline src/pybind/mgr/dashboard_v2/README.rst:97]: Fixed by https://github.com/openattic/ceph/pull/82/files#diff-b82e11902c1a4d3d8ad2c8d811c328c4R28

File: src/pybind/mgr/dashboard_v2/README.rst [hunk 2]
```diff
@@ -71,30 +71,80 @@ Unit Testing and Linting
...
+  $ tox -e ceph-cluster-start
+
+The command above uses ``vstart.sh`` script to start a Ceph cluster and
+automatically enables the ``dashboard_v2`` module, and configures its cherrypy
+web server to listen in port ``9865``.
+
+After starting the Ceph cluster we can run our test file using ``py.test``
+like this::
+
+  $ PATH=../../../../build/bin:$PATH DASHBOARD_V2_PORT=9865 UNITTEST=true py.test -s tests/test_mycontroller.py
```
Comment (sebastian-philipp) [inline src/pybind/mgr/dashboard_v2/README.rst:137]: Remove `PATH`

File: src/pybind/mgr/dashboard_v2/README.rst [hunk 3]
```diff
@@ -310,36 +416,62 @@ can be used:
...
+          self.rbd = rbd.RBD()
 
-  class RbdImagesTest(ControllerTestCase):
-      @mock.patch('rbd.RBD.list')
-      def test_list(self, rbd_list_mock):
-          rbd_list_mock.return_value = ['img1', 'img2']
-          self._get('/api/rbdimages')
-          self.assertJsonBody([{'name': 'img1'}, {'name': 'img2'}])
+      @ViewCache()
+      def image_list(self, pool_name):
```
Comment (sebastian-philipp) [inline src/pybind/mgr/dashboard_v2/README.rst:449]: `image_list` can be `static`

File: src/pybind/mgr/dashboard_v2/README.rst [hunk 4]
```diff
@@ -310,36 +416,62 @@ can be used:
...
+the function.
+The ``@ViewCache`` accepts a timeout argument in seconds that specifies the
+time to wait for the function to return. The default timeout is 5 seconds.
+
+The state flag can have four different values:
+
+* ``VALUE_OK = 0``: the returned data is fresh.
+* ``VALUE_STALE = 1``: the returned data might be stale.
+* ``VALUE_NODE = 2``: no data could be returned.
+* ``VALUE_EXCEPTION = 3``: an exception was raised while running the function.
```
Comment (sebastian-philipp) [inline src/pybind/mgr/dashboard_v2/README.rst:476]: Which integer value they exactly correspond to is somehow irrelevant.

File: src/pybind/mgr/dashboard_v2/README.rst (hunk 1 above)
Comment (rjfd) [reply src/pybind/mgr/dashboard_v2/README.rst:97]: great!

File: src/pybind/mgr/dashboard_v2/README.rst (hunk 4 above)
Comment (rjfd) [reply src/pybind/mgr/dashboard_v2/README.rst:476]: It's relevant for the frontend guys that consume this

File: src/pybind/mgr/dashboard_v2/README.rst (hunk 3 above)
Comment (rjfd) [reply src/pybind/mgr/dashboard_v2/README.rst:449]: in this particular case no, because we are using a instance variable like `rbd`. Also, it's just an example.

**Review by jecluis — CHANGES_REQUESTED** [review]
commit title for `mgr/dashboard_v2: Updated README with more developer notes ...` is too long and overflows. I'd reword it as
`mgr/dashboard_v2: README: dev notes on controllers` or something similar.

**Comment by LenzGr** [comment]
Changes have been merged into PR #95
